src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java

```diff
@@ -0,0 +1,115 @@
+
+package edu.ucla.library.prl.harvester.services;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import edu.ucla.library.prl.harvester.Institution;
+import edu.ucla.library.prl.harvester.MessageCodes;
+
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Optional;
+
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+
+import com.google.i18n.phonenumbers.NumberParseException;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.Phonenumber.PhoneNumber;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.vertx.config.ConfigRetriever;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.MessageConsumer;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.serviceproxy.ServiceBinder;
+
+/**
+ * Tests {@linkHarvestScheduleStoreService}.
+ */
+@ExtendWith(VertxExtension.class)
+@TestInstance(Lifecycle.PER_CLASS)
+public class HarvestScheduleStoreServiceIT {
+
+    /**
+     * The schedule store test's logger.
+     */
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(HarvestScheduleStoreServiceIT.class, MessageCodes.BUNDLE);
+
+    private static final PhoneNumberUtil PHONE_NUMBER_UTIL = PhoneNumberUtil.getInstance();
+
+    private MessageConsumer<?> myHarvestScheduleStoreService;
+
+    private HarvestScheduleStoreService myScheduleStoreProxy;
+
+    /**
+     * Sets up the test service.
+     *
+     * @param aVertx A A Vert.x instance
+     * @param aContext A test context
+     */
+    @BeforeAll
+    public final void setUp(final Vertx aVertx, final VertxTestContext aContext) {
+        final ConfigRetriever cr = ConfigRetriever.create(aVertx);
+
+        cr.getConfig().compose(config -> {
+            return Future.succeededFuture(HarvestScheduleStoreService.create(aVertx, config));
```

Comment:
Since it's not async, the call to HarvestScheduleStoreService#create could be moved into the onSuccess method.

Reply:
I'm probably being dense... but I don't see how to pass the config object generated by `cr.getConfig()` to the `onSuccess` clause.

Reply:
Not at all.

```java
cr.getConfig().compose(config -> {
    return Future.succeededFuture(HarvestScheduleStoreService.create(aVertx, config));
}).onSuccess(result -> {
    // ...
});
```

is essentially equivalent to

```java
cr.getConfig().onSuccess(config -> {
    var service = HarvestScheduleStoreService.create(aVertx, config);
    // ...
});
```

because in the former, a call to compose will _always_ result in "success", therefore it will _always_ be followed by a call to onSuccess, so it can be rewritten as a single call to onSuccess (passing in the config instead of the service).

Reply:
Haven't confirmed this works, but possibly:
```
ConfigRetriever.create(aVertx).getConfig().onSuccess(config -> {
    final HarvestScheduleStoreService service = HarvestScheduleStoreService.create(aVertx, config);
    final ServiceBinder binder = new ServiceBinder(aVertx).setAddress(HarvestScheduleStoreService.ADDRESS);

    myHarvestScheduleStoreService = binder.register(HarvestScheduleStoreService.class, service);
    myScheduleStoreProxy = HarvestScheduleStoreService.createProxy(aVertx);

    aContext.completeNow();
}).onFailure(aContext::failNow);
```

Reply:
I'd put in a +1 for variable names more descriptive than `sb` (which is also frequently used for StringBuffer) and `cr` too.

Reply:
Ooops, obviously hadn't refreshed my screen to see Mark had already answered :-)